Services/User/classes/class.ilObjUserFolderGUI.php

```diff
@@ -1941,9 +1942,13 @@ public function saveGeneralSettingsObject(): void
                     'user_delete_own_account_email',
                     $this->form->getInput('user_own_account_email')
                 );
+                $this->settings->set(
+                    'dpro_withdrawal_usr_deletion',
+                    (string) ((int) $this->form->getInput('dpro_withdrawal_usr_deletion'))
+                );
                 $this->settings->set(
                     'tos_withdrawal_usr_deletion',
-                    (string) ($this->form->getInput('tos_withdrawal_usr_deletion'))
+                    (string) ((int) $this->form->getInput('tos_withdrawal_usr_deletion'))
```

Comment:
I assume you expecting to return `0` or `1` here therefore i would encourage your to transform the input instzead of casting it to ensure unity, like:
`($this->form->getInput('tos_withdrawal_usr_deletion') === '1') ? '1' : '0'`